python/ray/train/tests/test_huggingface_gpu.py

```diff
@@ -36,9 +36,13 @@ def create_checkpoint():
 @pytest.mark.parametrize("device", [None, 0])
 def test_predict_batch(ray_start_4_cpus, caplog, batch_type, device):
     checkpoint = create_checkpoint()
-    predictor = BatchPredictor.from_checkpoint(
-        checkpoint, HuggingFacePredictor, task="text-generation"
-    )
+    kwargs = {}
+    if device:
+        kwargs["device"] = device
+    with caplog.at_level(logging.WARNING):
+        predictor = BatchPredictor.from_checkpoint(
+            checkpoint, HuggingFacePredictor, task="text-generation", **kwargs
```

Comment:
can we add a test where `device` is not explicitly set, making sure that GPU is still used?

Reply:
That's tested through `device=None`